testing/src/main/java/io/grpc/internal/testing/TestUtils.java

```diff
@@ -189,10 +189,11 @@ public static SSLSocketFactory newSslSocketFactoryForCa(Provider provider,
     KeyStore ks = KeyStore.getInstance(KeyStore.getDefaultType());
     ks.load(null, null);
     CertificateFactory cf = CertificateFactory.getInstance("X.509");
-    X509Certificate cert = (X509Certificate) cf.generateCertificate(
-        new BufferedInputStream(new FileInputStream(certChainFile)));
-    X500Principal principal = cert.getSubjectX500Principal();
-    ks.setCertificateEntry(principal.getName("RFC2253"), cert);
+    try (BufferedInputStream in = new BufferedInputStream(new FileInputStream(certChainFile))) {
```

Comment:
We can't use try-with-resources currently, because of our Android support (it's complicated). Use a try-finally instead:

```
BufferedInputStream in = new BufferedInputStream(new FileInputStream(certChainFile));
try {
  ...
} finally {
  in.close();
}
```

Reply:
Ah, that's too bad. Done.